src/components/Dashboard/panels/ANJForm.js

```diff
@@ -1,38 +1,49 @@
-import React, { useCallback, useMemo, useState } from 'react'
-import { Button, Field, TextInput, useSidePanelFocusOnReady } from '@aragon/ui'
+import React, { useCallback, useState } from 'react'
+import {
+  Button,
+  Field,
+  GU,
+  Info,
+  TextInput,
+  useSidePanelFocusOnReady,
+  useTheme,
+} from '@aragon/ui'
 
-import { parseUnits, formatUnits, bigNum } from '../../../lib/math-utils'
+import { parseUnits, formatUnits } from '../../../lib/math-utils'
 import { useCourtConfig } from '../../../providers/CourtConfig'
 
 const NO_ERROR = Symbol('NO_ERROR')
 
-const MAX_INPUT_DECIMAL_BASE = 3
-
 const ANJForm = React.memo(function ANJForm({
   actionLabel,
+  maxAmount,
   onDone,
   onSubmit,
   validateForm,
   errorToMessage,
 }) {
   const [amount, setAmount] = useState({ value: '0', error: NO_ERROR })
   const { anjToken } = useCourtConfig()
+  const theme = useTheme()
   const inputRef = useSidePanelFocusOnReady()
 
-  const minStep = useMemo(
-    () =>
-      formatUnits(bigNum(1), {
-        digits: Math.min(anjToken.decimals, MAX_INPUT_DECIMAL_BASE),
-      }),
-    [anjToken.decimals]
-  )
-
   // Change amount handler
   const handleAmountChange = useCallback(event => {
     const newAmount = event.target.value
     setAmount(amount => ({ ...amount, value: newAmount }))
   }, [])
```

Comment:
I think we should store two values here: `event.target.value` to update the field in the next render, and a filtered version of it as a BN.js number.

If the value can’t be converted into a BN.js number, we could have a value representing its impossible state. For example, if the user types “abc”, we could use `bigNum(-1)` or `null`.

Another option would be to do this on submit, but we might want to know the parsed value before that (e.g. to inform that the value is higher than the max).

Reply:
Ah yes, good idea. I also ended up using the common form errors on the `ANJForm` and removing them from the parent panel.